e2e/404.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { test, expect, type Page } from '@playwright/test';
+import translations from '../client/i18n/locales/english/translations.json';
+
+let page: Page;
+
+test.beforeAll(async ({ browser }) => {
+  page = await browser.newPage();
+  await page.goto('/nonexistent-url');
+});
+
+test.afterAll(async () => {
+  await page.close();
+});
+
+test.describe('404 Page Tests', () => {
+  test('The page should have correct heading', async () => {
```

Comment:
```suggestion
  test('The page should have correct heading', async ({ page }) => {
```

We can get `page` directly from the function args.